src/EnsembleKalmanProcess.jl

```diff
@@ -336,4 +337,336 @@ function update_ensemble!(ekp::EnsembleKalmanProcess{FT, IT, Sampler{FT}}, g_in:
+    uu_cov::Vector{Array{FT, 2}}
+    "a vector of arrays of size N_y containing the predicted observation (in each uki iteration a new array of predicted observation is added)"
+    obs_pred::Vector{Array{FT, 1}}
+    "weights in UKI"
+    c_weights::Array{FT, 1}
+    mean_weights::Array{FT, 1}
+    cov_weights::Array{FT, 1}
+    "covariance of the artificial evolution error"
+    Σ_ω::Array{FT, 2}
+    "covariance of the artificial observation error"
+    Σ_ν::Array{FT, 2}
+    "regularization parameter"
+    α_reg::FT
+    "regularization vector"
+    r::Array{FT, 1}
+    "update frequency"
+    update_freq::IT
+    "current iteration number"
+    iter::IT
+end
+
+# outer constructors
+function EnsembleKalmanProcess(
+    obs_mean::Array{FT, 1},
+    obs_noise_cov::Array{FT, 2},
+    process::Unscented{FT, IT};
+    Δt=FT(1)) where {FT<:AbstractFloat, IT<:Int}
+
+#initial parameters stored as columns
+init_params = [DataContainer(update_ensemble_prediction!(process), data_are_columns=true)]
+# Number of parameter
+N_u = length(process.u_mean[1])
+# ensemble size
+N_ens = 2N_u + 1 #stored with data as columns
+#store for model evaluations
+g=[] 
+# error store
+err = FT[]
+# timestep store
+Δt = Array([Δt])
+
+EnsembleKalmanProcess{FT, IT, Unscented}(init_params, obs_mean, obs_noise_cov, N_ens, g,
+          err, Δt, process)
+end
+
+
+
+"""
+EnsembleKalmanProcess Constructor 
+u0_mean::Array{FT} : prior mean
+uu0_cov::Array{FT, 2} : prior covariance
+obs_mean::Array{FT,1} : observation 
+obs_noise_cov::Array{FT, 2} : observation error covariance
+α_reg::FT : regularization parameter toward u0 (0 < α_reg <= 1), default should be 1, without regulariazion
+update_freq::IT : set to 0 when the inverse problem is not identifiable, 
+                  namely the inverse problem has multiple solutions, 
+                  the covariance matrix will represent only the sensitivity of the parameters, 
+                  instead of posterior covariance information;
+                  set to 1 (or anything > 0) when the inverse problem is identifiable, and 
+                  the covariance matrix will converge to a good approximation of the 
+                  posterior covariance with an uninformative prior.
+                  
+"""
+function Unscented(
+    u0_mean::Array{FT, 1}, 
+    uu0_cov::Array{FT, 2},
+    obs_mean::Array{FT,1},
+    obs_noise_cov::Array{FT, 2},
+    α_reg::FT,
+    update_freq::IT;
+    modified_uscented_transform::Bool = true) where {FT<:AbstractFloat, IT<:Int}
+    
+    N_u = size(u0_mean,1)
+    N_y = size(obs_mean, 1)
+    # ensemble size
+    N_ens = 2*N_u+1
+    
+    
+    c_weights = zeros(FT, N_u)
+    mean_weights = zeros(FT, N_ens)
+    cov_weights = zeros(FT, N_ens)
+    
+    # todo parameters λ, α, β
+    
+    κ = 0.0
+    β = 2.0
+    α = min(sqrt(4/(N_u + κ)), 1.0)
+    λ = α^2*(N_u + κ) - N_u
```

Comment:
The parameters `kappa` `beta` are essentially magic numbers. they should either be replaced with actually `0.0` and `2.0` or they should be parameters - perhaps with these default values (that can be changed by a user, and have some brief explanation).

Reply:
Updated